include/swift/Option/Options.td

```diff
@@ -293,9 +293,9 @@ def profile_stats_entities: Flag<["-"], "profile-stats-entities">,
 def emit_dependencies : Flag<["-"], "emit-dependencies">,
   Flags<[FrontendOption, NoInteractiveOption, DoesNotAffectIncrementalBuild]>,
   HelpText<"Emit basic Make-compatible dependencies files">;
-def track_system_dependencies : Flag<["-"], "track-system-dependencies">,
```

Comment:
I don't think we can remove this flag so easily, that would be a breaking change.

Reply:
I understand the concern. However, it seems the build system doesn't specify this flag at all. Also, this driver flag never populates to front-end invocation flags, meaning even if some one specifies the flag, it doesn't change any behavior of the compiler. It should be pretty safe to remove the flag.

Reply:
> However, it seems the build system doesn't specify this flag at all.

I don't understand which build system are you referring to. Maybe we are not using it but someone external could very well be using it?

> Also, this driver flag never populates to front-end invocation flags, meaning even if some one specifies the flag, it doesn't change any behavior of the compiler.

In that case, we should have the doc comment say something like "[Deprecated] We track system dependencies by default, so this flag doesn't do anything. If you want to avoid tracking system dependencies, use -skip-system-dependencies.". We could even add a warning saying that the flag is a no-op and asking the user to remove it.

I don't see what we gain by removing the flag -- we're potentially breaking someone's code for (IMO) no benefit at all.